### DISCUSSION TITLE
Custom font family in Kotlin project

### DISCUSSION THREAD

**Discussion by PoslavskySV** [JetBrains/lets-plot-kotlin#124] [General]
Dear Colleagues,

thank you very much for excellent software. We use lets-plot directly in Kotlin (not in Jupyter) to generate plots. Is there a way to use custom fonts: say I have a ttf font in the resources - how I can make lets-plot see it when using `geomText(family = "myFontFamily")` / axes labels etc?

Thanks!
Stanislav

**Comment by alshan** [comment]
Hi Stanislav, 
font on axis, legends and titles is not yet customizable unfortunately. 

Does `geomText(family=..)` work for you? I believe it should.

**Comment by PoslavskySV** [comment]
Regarding `geomText(family=..)` - I have a custom `.ttf` font file (not a system-wide), so I believe I need to make it recognizable by lets-plot?

**Comment by alshan** [comment]
Try to register your custom font in the platform, either by using `GraphicsEnvironment.registerFont()` or via [Font Configuration Files](https://docs.oracle.com/javase/9/intl/font-configuration-files.htm#JSINT-GUID-1F5136AF-907B-4C64-8657-73D953D7B912).

Once your font family is available in runtime you must (in theory) be able to use it in `geom_text()`.

**Comment by PoslavskySV** [comment]
`GraphicsEnvironment` etc unfortunately does not work when you want to export graphics in PDF format. So finally I came up with the following solution:

https://stackoverflow.com/a/72807651/946635

Using this, one can specify font family as usual (`geomText(family="IBM Plex Mono")`).
